Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -155,83 +161,35 @@ You'll end up with a structure that resembles the original Android Studio projec
-
-![User Input Check](Images/Chapters/0x05c/check_input.jpg)
-
-Right-click the class name—the first `a` in `a.a`—and select Refactor->Rename from the drop-down menu (or press Shift-F6). Change the class name to something that makes more sense given what you know about the class so far. For example, you could call it "Validator" (you can always revise the name later). `a.a` now becomes `Validator.a`. Follow the same procedure to rename the static method `a` to `check_input`.
-
-![Refactored class and method names](Images/Chapters/0x05c/refactored.jpg)
-
-Congratulations—you just learned the fundamentals of static analysis! It is all about theorizing, annotating, and gradually revising theories about the analyzed program until you understand it completely—or, at least, well enough for whatever you want to achieve.
-
-Next, Ctrl+click (or Command+click on Mac) on the `check_input` method. This takes you to the method definition. The decompiled method looks like this:
-
-```java
-    public static boolean check_input(String string) {
-        byte[] arrby = Base64.decode((String)"5UJiFctbmgbDoLXmpL12mkno8HT4Lv8dlat8FxR2GOc=", (int)0);
-        byte[] arrby2 = new byte[]{};
-        try {
-            arrby = sg.vantagepoint.a.a.a(Validator.b("8d127684cbc37c17616d806cf50473cc"), arrby);
-            arrby2 = arrby;
-        }sa
-        catch (Exception exception) {
-            Log.d((String)"CodeCheck", (String)("AES error:" + exception.getMessage()));
-        }
-        if (string.equals(new String(arrby2))) {
-            return true;
-        }
-        return false;
-    }
-```
-
-So, you have a Base64-encoded String that's passed to the function `a` in the package `sg.vantagepoint.a.a` (again, everything is called `a`) along with something that looks suspiciously like a hex-encoded encryption key (16 hex bytes = 128bit, a common key length). What exactly does this particular `a` do? Ctrl-click it to find out.
-
-```java
-public class a {
-    public static byte[] a(byte[] object, byte[] arrby) {
-        object = new SecretKeySpec((byte[])object, "AES/ECB/PKCS7Padding");
-        Cipher cipher = Cipher.getInstance("AES");
-        cipher.init(2, (Key)object);
-        return cipher.doFinal(arrby);
-    }
-}
-```
-
-Now you're getting somewhere: it's simply standard AES-ECB. Looks like the Base64 string stored in `arrby1` in `check_input` is a ciphertext. It is decrypted with 128bit AES, then compared with the user input. As a bonus task, try to decrypt the extracted ciphertext and find the secret value!
-
-A faster way to get the decrypted string is to add dynamic analysis—we'll revisit UnCrackable Level 1 later to show how, so don't delete the project yet!
```

Comment:
where did this go to :) ?

Reply:
Now it's in "Reviewing Decompiled Java Code"